spark-bigquery-dsv2/spark-3.1-bigquery/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryTableProvider.java

```diff
@@ -15,33 +15,88 @@
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.SupportsCatalogOptions;
 import org.apache.spark.sql.connector.catalog.Table;
 import org.apache.spark.sql.connector.catalog.TableProvider;
 import org.apache.spark.sql.connector.expressions.Transform;
 import org.apache.spark.sql.types.StructType;
 import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
-public class BigQueryTableProvider extends BaseBigQuerySource implements TableProvider {
+public class BigQueryTableProvider extends BaseBigQuerySource
+    implements TableProvider, SupportsCatalogOptions {
 
+  private static final String DEFAULT_CATALOG_NAME = "bigquery";
+  private static final String DEFAULT_CATALOG = "spark.sql.catalog." + DEFAULT_CATALOG_NAME;
   private static final Transform[] EMPTY_TRANSFORM_ARRAY = {};
 
   @Override
   public StructType inferSchema(CaseInsensitiveStringMap options) {
-    return getTable(null, EMPTY_TRANSFORM_ARRAY, options).schema();
+    return getBigQueryTableInternal(options).schema();
   }
 
   @Override
   public Table getTable(
       StructType schema, Transform[] partitioning, Map<String, String> properties) {
-    Injector injector = InjectorFactory.createInjector(schema, properties);
-    BigQueryTable table = new BigQueryTable(injector, schema);
-    return table;
+    setupDefaultSparkCatalog(SparkSession.active());
+    return getBigQueryTableInternal(schema, properties);
+  }
+
+  private BigQueryTable getBigQueryTableInternal(Map<String, String> properties) {
+    return getBigQueryTableInternal(null, properties);
+  }
+
+  private BigQueryTable getBigQueryTableInternal(
+      StructType schema, Map<String, String> properties) {
+    try {
+      Injector injector =
+          InjectorFactory.createInjector(schema, properties, /* tableIsMandatory */ true);
+      BigQueryTable table = BigQueryTable.fromConfigurationAnsSchema(injector, schema);
+      return table;
+    } catch (NoSuchTableException e) {
+      throw new BigQueryConnectorException("Table was not found", e);
+    }
   }
 
   @Override
   public boolean supportsExternalMetadata() {
     return true;
   }
+
+  @Override
+  public Identifier extractIdentifier(CaseInsensitiveStringMap options) {
+    ImmutableMap<String, String> globalOptions =
+        ImmutableMap.copyOf(mapAsJavaMap(SparkSession.active().conf().getAll()));
+    return new BigQueryIdentifier(
+        BigQueryConfigurationUtil.parseSimpleTableId(globalOptions, options));
+  }
+
+  @Override
+  public String extractCatalog(CaseInsensitiveStringMap options) {
+    setupDefaultSparkCatalog(SparkSession.active());
+    return DEFAULT_CATALOG_NAME;
+  }
+
+  private static void setupDefaultSparkCatalog(SparkSession spark) {
+    if (spark.conf().contains(DEFAULT_CATALOG)) {
+      return;
+    }
+    ImmutableMap<String, String> config =
+        ImmutableMap.of(
+            "type", "hive",
```

Comment:
Is `hive` correct? Is it because the catalog leverages the hive metastore or something?